### ISSUE TITLE
Add pytest and fixtures to documentation

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#1507]
We should add a section to the documentation with information on writing tests. Much of this could be copied from #1502 and mixed with a little generic pytest documentation.

This should also include information on the available fixtures like the [bare `opsdroid` one](https://github.com/opsdroid/opsdroid/blob/7c765997b514d3c0f3047bb6fc09da4a493aed16/opsdroid/conftest.py#L8). It should be possible to set this up with [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) and then ensure that the fixtures have well written docstrings.

I suggest the best place for this would be in the `contributing.md` file.

**Comment by Cadair** [comment]
We should also ensure they are easily importable into other packages so that downstream projects can use the fixtures.

**Comment by jacobtomlinson** [comment]
Yeah definitely! One thing I was trying to work out was if you had some matrix foxtures in a `conftest.py` file in`opsdroid/connectors/matrix/tests` you would import it as `opsdroid.connectors.matrix.tests.conftest` right?

But if we have root level tests in opsdroid they would need to be in `opsdroid/conftest.py` which would mean an import of `opsdroid.conftest`. But really the import would be better as `opsdroid.tests.conftest`. Or maybe even `opsdroid.tests.fixtures`.

Do you have thoughts on how we structure things? Should we have `fixtures.py` files which are imported by the relevant `conftest.py` file?

**Comment by jacobtomlinson** [comment]
After writing that and reading back I think having `fixtures.py` files within `tests/` directories which are imported by the `conftest.py` files fixes the problem I mentioned and also makes a nice import for others.

I have added the tests as an exclude to the `MANIFEST.in` file, maybe we should revert that.
